### PR TITLE
Add cGAN_Tensorflow

### PR DESCRIPTION
**Pull request by bansal-vansh**
Please review..

### CODE REVIEW

File: generative_models/cGAN_Tensorflow/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+"""dataloader.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1tsiDWd3a13JZxEl6k7OoByHW8ync_nFR
+"""
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/dataloader.py]: remove this lines

File: generative_models/cGAN_Tensorflow/main.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+"""main.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1QOK6bAyJhvJRVJRws1JQMkss6RNJgY5r
+"""
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/main.py]: remove this lines

File: generative_models/cGAN_Tensorflow/models.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""model.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/103GaSOVBmoWrhmKtw4BOcC8Tmshzt1z6
+"""
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/models.py]: remove this lines

File: generative_models/cGAN_Tensorflow/utils.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+"""utils.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/12y1Oih5uRwdenPwJPMzNWjhuJjz5qd7g
+"""
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/utils.py]: remove this lines

File: generative_models/cGAN_Tensorflow/main.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    args = parser.parse_args()
+
+    return args
+
+args = parse_args()
+
+epochs = args.epochs
+dataset = args.dataset
+batch_size = args.batch_size
+outdir = args.outdir + dataset + "/"
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/main.py]: maybe add this as parser arg, and always use os.path.join(...) to join path.

File: generative_models/cGAN_Tensorflow/readme.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+## Usage
+```bash
+$ python3 main.py --dataset 'mnist' --epochs 40
+```
+> **_NOTE:_** on Colab Notebook use following command:
+```python
+!git clone link-to-repo
+%run main.py --dataset 'mnist' --epochs 40 
+```
+
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/readme.md]: Add help log section here

**Comment by nirmal-suthar** [comment]
- Rename readme.md -> README.md
- Add some images of results in your README.md, it will make readme more presentable.

File: generative_models/cGAN_Tensorflow/main.py [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    args = parser.parse_args()
+
+    return args
+
+args = parse_args()
+
+epochs = args.epochs
+dataset = args.dataset
+batch_size = args.batch_size
+outdir = os.path.join(args.outdir, dataset, "")
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/main.py]:
```suggestion
outdir = os.path.join(args.outdir, dataset)
```

File: generative_models/cGAN_Tensorflow/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+The optimization reduces to the following min max condition:
+<img src="https://miro.medium.com/max/1400/1*l2tSqFN0Afwizm4LgalCGg.png">
+
+Occasionally, we'll try to fool the discriminator by pretending that the synthetic data is real with a label 1.0. By maximizing with respect to (𝐷), the optimizer sends gradient updates to the discriminator parameters to consider this synthetic data as real. At the same time, by minimizing with respect to (𝐺), the optimizer will train the generator's parameters on how to trick the discriminator.
+
+## Results on the MNIST dataset
+Here is the loss graph
+> **_NOTE:_** For this graph we have trained the model for 30 epochs only.Accuracy will improve significantly if the model is trained for more epochs.
+
+![graph](https://raw.githubusercontent.com/vanshbansal1505/model-zoo/master/generative_models/cGAN_Tensorflow/assets/loss%20plot.png)
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/README.md]:
```suggestion
![graph](./assets/loss_plot.png)
```

File: generative_models/cGAN_Tensorflow/README.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+This is the final gif generated by the generator for a randomly generated noise and label.
+
+![gif](https://raw.githubusercontent.com/vanshbansal1505/model-zoo/master/generative_models/cGAN_Tensorflow/assets/cgan_mnist.gif)
```
Comment (nirmal-suthar) [inline generative_models/cGAN_Tensorflow/README.md]:
```suggestion
![gif](./assets/cgan_mnist.gif)
```

File: generative_models/cGAN_Tensorflow/README.md (hunk 8 above)
Comment (nirmal-suthar) [reply generative_models/cGAN_Tensorflow/README.md]: maybe rename "loss plot.png" file to remove space in its name.

**Review by nirmal-suthar — APPROVED** [review]
